spyder/plugins/updatemanager/widgets/update.py

```diff
@@ -514,12 +498,30 @@ def _start_major_install(self):
             ]
             for program in programs:
                 if is_program_installed(program['cmd']):
-                    cmd = [program['cmd'], program['exe-opt']] + sub_cmd
+                    cmd = [program['cmd'], program['exe-opt']]
                     break
+        cmd.append(self.installer_path)
+
+        env = os.environ.copy()
+        if sys.platform == "darwin":
+            # PKG installers will not pass environment variables in the GUI.
+            # To communicate whether to start Spyder, create dummy file.
+            no_start_file = osp.join(
+                osp.dirname(self.installer_path), "no-start-spyder"
+            )
+            if self.restart_spyder and osp.exists(no_start_file):
+                os.remove(no_start_file)
+            if not self.restart_spyder:
+                open(no_start_file, 'w').close()
+        else:
+            # EXE and SH installers will pass environment variables
+            env.update({"START_SPYDER": str(self.restart_spyder)})
+
+        logger.debug(f"Restart Spyder after install: {self.restart_spyder}")
 
         logger.debug(f"""Update command: "{' '.join(cmd)}" """)
 
-        subprocess.Popen(' '.join(cmd), shell=True)
+        subprocess.Popen(' '.join(cmd), shell=True, env=env)
```

Comment:
* So for major updates now we run the installers directly by calling them in this `subprocess`, right?
* If that's the case, please add clarifying a comment above this line to mention that. But if I'm mistaken, please also add a comment to explain how major updates are going to work from now on.

Reply:
I added clarifying language to the docstring for `_start_major_install` and `_start_updater`.

Reply:
Yes, for major updates we now run the installers directly in the subprocess rather than running an intermediate sh/bat script.